### ISSUE TITLE
Minimize action: windows order not preserved when activating windows

### ISSUE THREAD

**Issue by franglais125** [micheleg/dash-to-dock#347]
Here is a video of the issue: https://youtu.be/wSrGP253t5I

When setting the option to "Minimize" with the left click (no Shift), windows don't always come up in the same order. This is not a major problem, but it can be a distraction when having a maximized window (see beginning of the video).

I prepared a patch to fix this [9fd9c2c0ec93b2a0554f51bbf23e2c36c6e09e0a]. It involves creating a second loop over the list of windows. The problem lies within activateAllWindows().

I don't know what the performance impact can be as I don't use this feature, and hence I don't know what sort of responsiveness an user can expect. From my limited testing, there was not much of a difference with and without the patch.

**Comment by micheleg** [comment]
Yes, you spotted a bug. I'm wondering, Is it due to the order in which the windows are hidden (which change their order in the internal list of windows the shell keeps) rather than the order in which we restore them?

**Comment by franglais125** [comment]
I actually tried that, but I don't think that that's the case. As you can see in the video, you can trigger the same behavior by focusing a window from another application (in the video I use the desktop, as I have icons on it).

I tried to change the order for the window activation, but I couldn't fix it. I tried quite a bit, until I realized that using 2 loops was working. Main.activateWindow() must be doing something to the order.
